salt/client/ssh/__init__.py

```diff
@@ -1393,6 +1393,26 @@ def categorize_shim_errors(self, stdout_bytes, stderr_bytes, retcode):
         perm_error_fmt = 'Permissions problem, target user may need '\
                          'to be root or use sudo:\n {0}'
 
+        python_mismatch_options = {
+            '3.x': (
+                'Depending on the Python version on the target, you need to' +
+                ' install Python2.7 compatible salt on origin to add support for Python2.7 targets or' +
+                ' install Python2.6 compatible salt on origin to add support for Python2.6 targets or' +
+                ' upgrade to Python==3.x on target'
+            ),
+            '2.7': (
+                'Depending on the Python version on the target, you need to' +
+                ' upgrade to Python3 on origin to match Python3 on target or' +
+                ' install Python2.7 compatible Salt on origin to add support for Python2.7 targets or' +
+                ' install Python2.6 compatible Salt on origin to add support for Python2.6 targets'
+            ),
+            '2.6': (
+                'Upgrade Python on origin to match the Python version available on target' +
+                ' and install the corresponding Salt package(Python2 compatible Salt or Python3 compatible Salt).'
+            ),
+            'default': 'Matching Python>=2.6 version needed both on origin and target.'
```

Comment:
Generally I am OK with this, except this will be like an info on the CLI. So maybe either rephrase this not to repeat the same thing twice, or make it as a list of actions (and less verbose). Thus calling `salt-ssh` would return something looking like this on the terminal:

```
Python version error. Your options:
- Install Salt for Python 2.7 on a Master
- Install Salt for Python 2.6 on a Master
- Upgrade to Python 3.x on the client machine
```

...etc. It would be easier to comprehend and quickly understand the problem. Like we do with Zypper. :wink: